src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2067,12 +2067,13 @@ private void LogEventArgsMismatches(ParameterInfo[] infos, object?[] args)
             {
                 Type pType = infos[i].ParameterType;
 
+                Type? argType = args[i]?.GetType();
+
                 // Checking to see if the Parameter types (from the Event method) match the supplied argument types.
-                // Fail if one of two things hold : either the argument type is not equal to the parameter type, or the 
-                // argument is null and the parameter type is non-nullable.
-                if ((args[i] != null && (args[i]!.GetType() != pType)) // TODO-NULLABLE: Indexer nullability tracked (https://github.com/dotnet/roslyn/issues/34644)
-                    || (args[i] == null && (!(pType.IsGenericType && pType.GetGenericTypeDefinition() == typeof(Nullable<>))))
-                    )
+                // Fail if one of two things hold : either the argument type is not equal or assignable to the parameter type, or the 
+                // argument is null and the parameter type is non-nullable or a ref type.
+                if (args[i] != null && !(argType!.Equals(pType) || pType.IsAssignableFrom(argType))
+                    || args[i] == null && !((pType.IsGenericType && pType.GetGenericTypeDefinition() == typeof(Nullable<>)) || !pType.IsValueType))
```

Comment:
This seems different than the comment? If I assume pType = class Foo and arg = null:
Comment suggests that we should fail the type match in this case:

```
argument is null and the parameter type is non-nullable or a ref type
True             and (                     True         or True       )
True
```

Code:


```
args[i] == null && !((pType.IsGenericType && pType.GetGenericTypeDefinition() == typeof(Nullable<>)) || !pType.IsValueType)
true            && !((false               && false)                                                  || !false)
true            && !( false                                                                          || true)
true            && !( true )
true            && false
false
```

Which answer is the intended one?

Reply:
Ah, good catch.  The code has the correct logic and I mistyped the comment.